src/main/java/org/jfree/chart/block/FlowArrangement.java

```diff
@@ -199,7 +199,6 @@ protected Size2D arrangeFN(BlockContainer container, Graphics2D g2,
                 }
                 else {
                     // start new row
```

Comment:
Should we not reinitialize `haveItemsInRow` to `false` here?

Reply:
The original code clears the `itemsInRow` and adds a element to the `itemsInRow` afterwards in the same code block.

So I think the reinitialization is not required.